src/http/one/FieldParser.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (C) 1996-2025 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_SRC_HTTP_ONE_FIELDPARSER_H
+#define SQUID_SRC_HTTP_ONE_FIELDPARSER_H
+
+#include "http/FieldParser.h"
+
+namespace Http
+{
+namespace One
+{
+class FieldParser : public Http::FieldParser
+{
+    MEMPROXY_CLASS(Http1::FieldParser);
+
+public:
+    /// parse an HTTP header field from the given tokenizer
+    void parseFieldLine(::Parser::Tokenizer &, const http_hdr_owner_type);
```

Comment:
Please convert this method into a class constructor (instead of allowing special FieldParser instances that have not parsed anything at al or half-valid instances that have not finished parsing).

Reply:
No. The requested change would lock this parser to only being capable of parsing HTTP/1.x MiME based message headers.   This is an RFC 9112 field-line parser, not a RFC 822 / RFC 2616 header parser.

Reply:
> The requested change would lock this parser to only being capable of parsing HTTP/1.x MiME based message headers.   This is an RFC 9112 field-line parser, not a RFC 822 / RFC 2616 header parser.

How would converting this method into constructor prevent this class from parsing RFC 9112 field-lines?

Reply:
My argument is the opposite. Your request would lock it to **only** being capable of handling RFC9112 binary syntax. Forcing an entirely different object to parse alternative syntax, eg structured header fields.

Also, it would force a new object allocation per field-line. Which is one of the performance bottlenecks the old parsers have. The `ParserNG` project is moving to reusable `Parser` objects to reduce allocation overheads.